atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraKeyValueServiceImpl.java

```diff
@@ -2074,9 +2090,19 @@ public ListenableFuture<Map<Cell, Value>> getAsync(TableReference tableRef, Map<
             return Futures.immediateFuture(ImmutableMap.of());
         }
 
-        return asyncKeyValueService
-                .map(asyncKvs -> asyncKvs.getAsync(tableRef, timestampByCell))
-                .orElseGet(() -> Futures.immediateFuture(this.get(tableRef, timestampByCell)));
+        try {
+            if (asyncKeyValueService.isValid()) {
+                return asyncKeyValueService.getAsync(tableRef, timestampByCell);
```

Comment:
This is a little gross. The alternative was to have a ReloadingCloseableContainer<Optional<AsyncKVS>>, but the dissenting opinion was a reloadable Optional was even more jank.